### ISSUE TITLE
Not functional installation of sage 9.7

### ISSUE THREAD

**Issue by Pascal-Ortiz** [conda-forge/sage-feedstock#80]
### Comment:

I installed sage 9.7 via conda-forge and my installation of sage does not work, as explained in this [post](https://ask.sagemath.org/question/64602/conda-forge-install-sagemath-97-fails/).

My OS is Ubuntu 18.04. First, I installed mamba using the instructions found [here](https://doc.sagemath.org/html/en/installation/conda.html#install-from-conda-forge).

Then, following the instruction [here](https://doc.sagemath.org/html/en/installation/conda.html#installing-all-of-sagemath-from-conda-not-for-development), I created the sage environment with this command line:

`mamba create -n sage sage python=3.9`

Then I activated it like this

`(base) po@po-ta:~$ conda activate sage`

Finally I ran sage:

`(sage) po@po-ta:~$ sage`

Here is the entire screen output I got:

```
(sage) po@po-ta:~$ sage
┌────────────────────────────────────────────────────────────────────┐
│ SageMath version 9.7, Release Date: 2022-09-19                     │
│ Using Python 3.9.13. Type "help()" for help.                       │
└────────────────────────────────────────────────────────────────────┘
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Warning: sage.all is not available; this is a limited REPL.        ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
sage: 
```

The sage installation is not functional as the warning explains.

**Comment by saraedum** [comment]
I cannot reproduce this.

Could you check that you are running the correct version of SageMath, i.e., what's the output of `which sage`.

Maybe there are some environment variables set that confuse Python here? What's the output of `export`?

Could you try running `sage` from a different directory, e.g., `cd /tmp/` then `sage`?

**Comment by poldolo** [comment]
I have the exact same issue on a stock OpenSuse Tumbleweed.

(sage) paolo@localhost:~> which sage
/home/paolo/anaconda3/envs/sage/bin/sage

declare -x ADDR2LINE="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-addr2line"
declare -x AR="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-ar"
declare -x AS="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-as"
declare -x AUDIODRIVER="pulseaudio"
declare -x BUILD="x86_64-conda-linux-gnu"
declare -x CC="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-cc"
declare -x CC_FOR_BUILD="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-cc"
declare -x CFLAGS="-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x CMAKE_ARGS="-DCMAKE_LINKER=/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-ld -DCMAKE_STRIP=/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-strip"
declare -x CMAKE_PREFIX_PATH="/home/paolo/anaconda3/envs/sage:/home/paolo/anaconda3/envs/sage/x86_64-conda-linux-gnu/sysroot/usr"
declare -x COLORFGBG="15;0"
declare -x COLORTERM="truecolor"
declare -x CONDA_BACKUP_HOST="localhost.localdomain"
declare -x CONDA_BACKUP_JAVA_HOME="/usr/lib64/jvm/jre-openjdk:-"
declare -x CONDA_BACKUP_JAVA_LD_LIBRARY_PATH=":-"
declare -x CONDA_BUILD_SYSROOT="/home/paolo/anaconda3/envs/sage/x86_64-conda-linux-gnu/sysroot"
declare -x CONDA_DEFAULT_ENV="sage"
declare -x CONDA_EXE="/home/paolo/anaconda3/bin/conda"
declare -x CONDA_PREFIX="/home/paolo/anaconda3/envs/sage"
declare -x CONDA_PREFIX_1="/home/paolo/anaconda3"
declare -x CONDA_PROMPT_MODIFIER="(sage) "
declare -x CONDA_PYTHON_EXE="/home/paolo/anaconda3/bin/python"
declare -x CONDA_SHLVL="2"
declare -x CONFIG_SITE="/usr/share/site/x86_64-pc-linux-gnu"
declare -x CPP="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-cpp"
declare -x CPPFLAGS="-DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x CPU="x86_64"
declare -x CSHEDIT="emacs"
declare -x CXX="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-c++"
declare -x CXXFILT="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-c++filt"
declare -x CXXFLAGS="-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x CXX_FOR_BUILD="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-c++"
declare -x DBUS_SESSION_BUS_ADDRESS="unix:path=/run/user/1000/bus"
declare -x DEBUGINFOD_URLS="https://debuginfod.opensuse.org/"
declare -x DEBUG_CFLAGS="-march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-all -fno-plt -Og -g -Wall -Wextra -fvar-tracking-assignments -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x DEBUG_CPPFLAGS="-D_DEBUG -D_FORTIFY_SOURCE=2 -Og -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x DEBUG_CXXFLAGS="-fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-all -fno-plt -Og -g -Wall -Wextra -fvar-tracking-assignments -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x DEBUG_FFLAGS="-fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include -fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-all -fno-plt -Og -g -Wall -Wextra -fcheck=all -fbacktrace -fimplicit-none -fvar-tracking-assignments -ffunction-sections -pipe"
declare -x DEBUG_FORTRANFLAGS="-fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include -fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-all -fno-plt -Og -g -Wall -Wextra -fcheck=all -fbacktrace -fimplicit-none -fvar-tracking-assignments -ffunction-sections -pipe"
declare -x DESKTOP_SESSION="/usr/share/xsessions/default"
declare -x DISPLAY=":0"
declare -x ELFEDIT="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-elfedit"
declare -x F77="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gfortran"
declare -x F90="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gfortran"
declare -x F95="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-f95"
declare -x FC="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gfortran"
declare -x FFLAGS="-fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x FORTRANFLAGS="-fopenmp -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /home/paolo/anaconda3/envs/sage/include"
declare -x FROM_HEADER=""
declare -x GCC="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gcc"
declare -x GCC_AR="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gcc-ar"
declare -x GCC_NM="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gcc-nm"
declare -x GCC_RANLIB="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gcc-ranlib"
declare -x GFORTRAN="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gfortran"
declare -x GPG_TTY="/dev/pts/8"
declare -x GPROF="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-gprof"
declare -x GTK2_RC_FILES="/etc/gtk-2.0/gtkrc:/home/paolo/.gtkrc-2.0:/home/paolo/.config/gtkrc-2.0"
declare -x GTK_IM_MODULE="cedilla"
declare -x GTK_MODULES="canberra-gtk-module"
declare -x GTK_RC_FILES="/etc/gtk/gtkrc:/home/paolo/.gtkrc:/home/paolo/.config/gtkrc"
declare -x GXX="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-g++"
declare -x G_BROKEN_FILENAMES="1"
declare -x G_FILENAME_ENCODING="@locale,UTF-8,ISO-8859-15,CP1252"
declare -x HISTSIZE="1000"
declare -x HOME="/home/paolo"
declare -x HOST="x86_64-conda-linux-gnu"
declare -x HOSTNAME="localhost.localdomain"
declare -x HOSTTYPE="x86_64"
declare -x INVOCATION_ID="470d4601be6f4fd196014a230ed1abd9"
declare -x JAVA_BINDIR="/usr/lib64/jvm/jre-openjdk/bin"
declare -x JAVA_HOME="/home/paolo/anaconda3/envs/sage"
declare -x JAVA_LD_LIBRARY_PATH="/home/paolo/anaconda3/envs/sage/lib/server"
declare -x JAVA_ROOT="/usr/lib64/jvm/jre-openjdk"
declare -x JOURNAL_STREAM="8:33750"
declare -x JRE_HOME="/usr/lib64/jvm/java-11-openjdk-11"
declare -x KDE_APPLICATIONS_AS_SCOPE="1"
declare -x KDE_FULL_SESSION="true"
declare -x KDE_SESSION_UID="1000"
declare -x KDE_SESSION_VERSION="5"
declare -x KONSOLE_DBUS_SERVICE=":1.48"
declare -x KONSOLE_DBUS_SESSION="/Sessions/5"
declare -x KONSOLE_DBUS_WINDOW="/Windows/1"
declare -x KONSOLE_VERSION="221203"
declare -x LANG="en_US.UTF-8"
declare -x LANGUAGE="en_US:it:fr"
declare -x LC_MEASUREMENT="it_IT.UTF-8"
declare -x LC_MONETARY="it_IT.UTF-8"
declare -x LC_NUMERIC="it_IT.UTF-8"
declare -x LC_TIME="it_IT.UTF-8"
declare -x LD="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-ld"
declare -x LDFLAGS="-Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--disable-new-dtags -Wl,--gc-sections -Wl,-rpath,/home/paolo/anaconda3/envs/sage/lib -Wl,-rpath-link,/home/paolo/anaconda3/envs/sage/lib -L/home/paolo/anaconda3/envs/sage/lib"
declare -x LD_GOLD="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-ld.gold"
declare -x LESS="-M -I -R"
declare -x LESSCLOSE="lessclose.sh %s %s"
declare -x LESSKEY="/usr/etc/lesskey.bin"
declare -x LESSOPEN="lessopen.sh %s"
declare -x LESS_ADVANCED_PREPROCESSOR="no"
declare -x LOGNAME="paolo"
declare -x LS_COLORS="no=00:fi=00:di=01;34:ln=00;36:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=41;33;01:ex=00;32:*.cmd=00;32:*.exe=01;32:*.com=01;32:*.bat=01;32:*.btm=01;32:*.dll=01;32:*.tar=00;31:*.tgz=00;31:*.arc=00;31:*.arj=00;31:*.taz=00;31:*.lha=00;31:*.lz4=00;31:*.lzh=00;31:*.lzma=00;31:*.tlz=00;31:*.txz=00;31:*.tzo=00;31:*.t7z=00;31:*.zip=00;31:*.z=00;31:*.Z=00;31:*.dz=00;31:*.gz=00;31:*.lrz=00;31:*.lz=00;31:*.lzo=00;31:*.xz=00;31:*.zst=00;31:*.tzst=00;31:*.bz2=00;31:*.bz=00;31:*.tbz=00;31:*.tbz2=00;31:*.tz=00;31:*.deb=00;31:*.rpm=00;31:*.jar=00;31:*.war=00;31:*.ear=00;31:*.sar=00;31:*.rar=00;31:*.alz=00;31:*.ace=00;31:*.zoo=00;31:*.cpio=00;31:*.7z=00;31:*.rz=00;31:*.cab=00;31:*.wim=00;31:*.swm=00;31:*.dwm=00;31:*.esd=00;31:*.asf=01;35:*.avi=01;35:*.bmp=01;35:*.cgm=01;35:*.dl=01;35:*.emf=01;35:*.flc=01;35:*.fli=01;35:*.flv=01;35:*.gif=01;35:*.gl=01;35:*.jpeg=01;35:*.jpg=01;35:*.m2v=01;35:*.m4v=01;35:*.mjpeg=01;35:*.mjpg=01;35:*.mkv=01;35:*.mng=01;35:*.mov=01;35:*.mp4=01;35:*.mp4v=01;35:*.mpeg=01;35:*.mpg=01;35:*.nuv=01;35:*.ogm=01;35:*.pbm=01;35:*.pcx=01;35:*.pgm=01;35:*.png=01;35:*.ppm=01;35:*.qt=01;35:*.rm=01;35:*.rmvb=01;35:*.svg=01;35:*.svgz=01;35:*.tga=01;35:*.tif=01;35:*.tiff=01;35:*.vob=01;35:*.webm=01;35:*.webp=01;35:*.wmv=01;35:*.xbm=01;35:*.xcf=01;35:*.xpm=01;35:*.xwd=01;35:*.yuv=01;35:*.ogv=01;35:*.ogx=01;35:*.aiff=00;32:*.ape=00;32:*.aac=00;32:*.au=00;32:*.flac=00;32:*.m4a=00;32:*.mid=00;32:*.midi=00;32:*.mka=00;32:*.mp3=00;32:*.mpc=00;32:*.ogg=00;32:*.ra=00;32:*.voc=00;32:*.wav=00;32:*.wma=00;32:*.wv=00;32:*.oga=00;32:*.opus=00;32:*.spx=00;32:*.xspf=00;32:"
declare -x LS_OPTIONS="-N --color=tty -T 0"
declare -x MACHTYPE="x86_64-suse-linux"
declare -x MAIL="/var/mail/paolo"
declare -x MANAGERPID="2169"
declare -x MANPATH="/usr/local/man:/usr/share/man"
declare -x MANPATHISSET="yes"
declare -x MATHJAX_DIR="/lib/python3.9/site-packages/notebook/static/components/MathJax"
declare -x MINICOM="-c on"
declare -x MORE="-sl"
declare -x NM="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-nm"
declare -x OBJCOPY="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-objcopy"
declare -x OBJDUMP="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-objdump"
declare -x OLDPWD="/home/paolo"
declare -x OSTYPE="linux"
declare -x PAGER="less"
declare -x PATH="/home/paolo/anaconda3/envs/sage/bin:/home/paolo/anaconda3/condabin:/home/paolo/.local/bin:/home/paolo/bin:/usr/local/bin:/usr/bin:/bin"
declare -x PROFILEHOME=""
declare -x PROFILEREAD="true"
declare -x PWD="/home/paolo"
declare -x PYTHONSTARTUP="/etc/pythonstart"
declare -x QEMU_AUDIO_DRV="pa"
declare -x QT_AUTO_SCREEN_SCALE_FACTOR="0"
declare -x QT_IM_MODULE="xim"
declare -x QT_IM_SWITCHER="imsw-multi"
declare -x RANLIB="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-ranlib"
declare -x READELF="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-readelf"
declare -x RSTUDIO_WHICH_R="/home/paolo/anaconda3/envs/sage/bin/R"
declare -x SAGE_PKGS="/home/paolo/anaconda3/envs/sage/var/lib/sage/installed"
declare -x SESSION_MANAGER="local/localhost.localdomain:@/tmp/.ICE-unix/3574,unix/localhost.localdomain:/tmp/.ICE-unix/3574"
declare -x SHELL="/bin/bash"
declare -x SHELL_SESSION_ID="7d4aeb839ede469d8e4cd179df05672b"
declare -x SHLVL="1"
declare -x SIZE="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-size"
declare -x SSH_ASKPASS="/usr/libexec/ssh/ssh-askpass"
declare -x STRINGS="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-strings"
declare -x STRIP="/home/paolo/anaconda3/envs/sage/bin/x86_64-conda-linux-gnu-strip"
declare -x SYSTEMD_EXEC_PID="3574"
declare -x TERM="xterm-256color"
declare -x USER="paolo"
declare -x VDPAU_DRIVER="va_gl"
declare -x VENDOR="suse"
declare -x WINDOWID="4194317"
declare -x WINDOWMANAGER="/usr/bin/startplasma-x11"
declare -x XAUTHLOCALHOSTNAME="localhost.localdomain"
declare -x XAUTHORITY="/run/user/1000/xauth_TdfuGV"
declare -x XCURSOR_SIZE="24"
declare -x XCURSOR_THEME="breeze_cursors"
declare -x XDG_CONFIG_DIRS="/home/paolo/.config/kdedefaults:/etc/xdg:/usr/local/etc/xdg:/usr/etc/xdg"
declare -x XDG_CURRENT_DESKTOP="KDE"
declare -x XDG_DATA_DIRS="/home/paolo/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share"
declare -x XDG_RUNTIME_DIR="/run/user/1000"
declare -x XDG_SEAT="seat0"
declare -x XDG_SEAT_PATH="/org/freedesktop/DisplayManager/Seat0"
declare -x XDG_SESSION_CLASS="user"
declare -x XDG_SESSION_DESKTOP="KDE"
declare -x XDG_SESSION_ID="1"
declare -x XDG_SESSION_PATH="/org/freedesktop/DisplayManager/Session0"
declare -x XDG_SESSION_TYPE="x11"
declare -x XDG_VTNR="7"
declare -x XKEYSYMDB="/usr/X11R6/lib/X11/XKeysymDB"
declare -x XMODIFIERS="@im=local"
declare -x XNLSPATH="/usr/share/X11/nls"
declare -x XSESSION_IS_UP="yes"
declare -x _CE_CONDA=""
declare -x _CE_M=""
declare -x _CONDA_PYTHON_SYSCONFIGDATA_NAME="_sysconfigdata_x86_64_conda_cos7_linux_gnu"
declare -x build_alias="x86_64-conda-linux-gnu"
declare -x host_alias="x86_64-conda-linux-gnu"

From /tmp same Warning.

Paolo.

****

**Comment by slel** [comment]
Maybe a locale problem? Might be worth checking whether these help:
```shell
export LANG=en_US.UTF-8
export LANGUAGE=en_US.UTF-8
export LC_ALL=en_US.UTF-8
export LC_CTYPE=en_US.UTF-8
```
Also, conda-forge now has SageMath 9.8.

**Comment by poldolo** [comment]
Thanks.

I would say we can close the issue as solved: started from scratch with python 3.10 and everything works fine now! SageMath 9.8 indeed.
